A command like /bin/sh ../../i586-poky-linux-libtool   --mode=install /usr/bin/install -c \
	gck-roots-store-standalone.la '/media/data1/builds/poky1/tmp/work/core2-poky-linux/gnome-keyring-2.26.1-r1/image/usr/lib/gnome-keyring/standalone/' fails (e.g. gnome-keyring or pulseaudio)

This is because libdir has a trailing slash which breaks the comparision.

RP 2/1/10

Merged a patch received from Gary Thomas <gary@mlbassoc.com>

Date: 2010/07/12
Nitin A Kamble <nitin.a.kamble@intel.com>

Adapted to libtool-2.4.6 file build-aux/ltmain.in
Jürgen Buchmüller <pullmoll@t-online.de>

--- a/build-aux/ltmain.in	2015-02-06 13:57:56.000000000 +0100
+++ b/build-aux/ltmain.in	2015-10-20 13:35:03.265959290 +0200
@@ -2363,8 +2363,15 @@
 	func_append dir "$objdir"
 
 	if test -n "$relink_command"; then
+	  # Strip any trailing slash from the destination.
+	  func_stripname '' '/' "$libdir"
+	  destlibdir=$func_stripname_result
+
+	  func_stripname '' '/' "$destdir"
+	  s_destdir=$func_stripname_result
+
 	  # Determine the prefix the user has applied to our future dir.
-	  inst_prefix_dir=`$ECHO "$destdir" | $SED -e "s%$libdir\$%%"`
+	  inst_prefix_dir=`$ECHO "X$s_destdir" | $Xsed -e "s%$destlibdir\$%%"`
 
 	  # Don't allow the user to place us outside of our expected
 	  # location b/c this prevents finding dependent libraries that
--- a/build-aux/ltmain.sh	2015-02-15 17:15:12.000000000 +0100
+++ b/build-aux/ltmain.sh	2015-10-20 23:26:41.674142100 +0200
@@ -4275,8 +4275,15 @@
 	func_append dir "$objdir"
 
 	if test -n "$relink_command"; then
+	  # Strip any trailing slash from the destination.
+	  func_stripname '' '/' "$libdir"
+	  destlibdir=$func_stripname_result
+ 
+	  func_stripname '' '/' "$destdir"
+	  s_destdir=$func_stripname_result
+ 
 	  # Determine the prefix the user has applied to our future dir.
-	  inst_prefix_dir=`$ECHO "$destdir" | $SED -e "s%$libdir\$%%"`
+	  inst_prefix_dir=`$ECHO "X$s_destdir" | $Xsed -e "s%$destlibdir\$%%"`
 
 	  # Don't allow the user to place us outside of our expected
 	  # location b/c this prevents finding dependent libraries that
